platform/pulp/plugin/changeset/iterator.py

```diff
@@ -0,0 +1,215 @@
+import itertools
+
+from logging import getLogger
+
+from django.db.models import Q
+
+from pulp.download import Download
+
+from .model import RemoteArtifact
+
+
+log = getLogger(__name__)
+
+
+class ContentIterator:
+    """
+    Iterate a generator of `RemoteContent` and replace associated content models
+    With models fetched from the DB (when found).
+    """
+
+    # Number of keys in each batched query.
+    BATCH = 1024
+
+    @staticmethod
+    def _batch_q(content):
+        """
+        Build a query for the specified batch of content.
+
+        Args:
+            content (tuple): A batch of content.  Each is: RemoteContent.
+
+        Returns:
+            Q: The built query.
+        """
+        query = Q()
+        for c in content:
+            query = query | Q(**c.key)
+        return query
+
+    def __init__(self, content):
+        """
+        Args:
+            content (generator): A generator of RemoteContent.
+        """
+        self.content = content
+
+    def _batch_content(self, size=BATCH):
```

Comment:
This looks familiar. ;)

https://github.com/pulp/pulp/blob/91a1e28c9e7d3dee418d5c7680dbf25c3e7adc63/server/pulp/plugins/util/misc.py#L14

It does tend to be generally useful, so having it in a "util" kind of module isn't a bad idea.